### ISSUE TITLE
In the case "Using pre-trained word embeddings in a Keras model", is this correct?

### ISSUE THREAD

**Issue by joshuia** [fchollet/keras#4515]
In the case "Using pre-trained word embeddings in a Keras model", I noticed that every document in 20news has the classification label. For example:

*****************************************************************************************************
Newsgroups: rec.autos
Path: cantaloupe.srv.cs.cmu.edu!magnesium.club.cc.cmu.edu!news.sei.cmu.edu!cis.ohio-state.edu!zaphod.mps.ohio-state.edu!darwin.sura.net!gatech!gsusgi1.gsu.edu!gsusgi1.gsu.edu!antkasx
From: antkasx@gsusgi2.gsu.edu (K A Sturrock)
Subject: Re: Viper Car Alarms
Message-ID: <antkasx.735868252@gsusgi1.gsu.edu>
Organization: Georgia State University
References: <C605Fr.CDo@cs.uiuc.edu>
Date: Mon, 26 Apr 1993 23:50:52 GMT
Lines: 26

u1815@cs.uiuc.edu (Ronald E Garnett) writes:

Ron> Viper also sells some fancy field disturbance sensor that
Ron> supposedly detects people approcahing the car....

******************************************************************************************************
It's the 103723 in rec.autos, at the first line you may notice that it's a document in newsgroup rec.autos. Does that affects the classification accuracy? That's why the accuracy up to 0.96? Is that a mistake?

**Comment by viksit** [comment]
Good question. Try using `sklearn`'s 20newsgroups data (which has the ability to remove headers etc built into the loading function) and try this? Curious to see what you find.

**Comment by viksit** [comment]
I did an experiment with that. Replace `texts`, `labels` and `labels_index` with the sklearn data (removing all headers and other information).

```bash
Indexing word vectors.
Found 400000 word vectors.
Processing text dataset
Found 18846 texts.
Found 409350 unique tokens.
Shape of data tensor: (18846, 1000)
Shape of label tensor: (18846, 20)
Preparing embedding matrix.
shape of embedding_matrix:  (20001, 100)
Training model.
Train on 15077 samples, validate on 3769 samples
Epoch 1/2
15077/15077 [==============================] - 28s - loss: 2.5867 - acc: 0.1597 - val_loss: 2.2434 - val_acc: 0.2314
Epoch 2/2
15077/15077 [==============================] - 28s - loss: 1.9416 - acc: 0.3414 - val_loss: 1.7384 - val_acc: 0.4150
```

With the original code,

```bash
Using Theano backend.
Using gpu device 0: GRID K520 (CNMeM is disabled, cuDNN 4007)
[I 2016-11-28 23:40:26,089 __main__] load spacy
[I 2016-11-28 23:40:32,448 __main__] done loading
Indexing word vectors.
Found 400000 word vectors.
Processing text dataset
Found 19997 texts.
Found 214909 unique tokens.
Shape of data tensor: (19997, 1000)
Shape of label tensor: (19997, 20)
Preparing embedding matrix.
shape of embedding_matrix:  (20001, 100)
Training model.
Train on 15998 samples, validate on 3999 samples
Epoch 1/2
15998/15998 [==============================] - 29s - loss: 1.4192 - acc: 0.5322 - val_loss: 0.2191 - val_acc: 0.9405
Epoch 2/2
15998/15998 [==============================] - 29s - loss: 0.1962 - acc: 0.9380 - val_loss: 0.1563 - val_acc: 0.9480
```

**Comment by joshuia** [comment]
Thank you for the good experiment! @viksit   
So the first one used the sklearn's 20newsgroups data, and the second used the original data, we may see the distinct differences in accuracy. 

And in the Chinese Keras Document, the interpreter added a link of experiment liked yours,  Rachel-Zhang used sklearn's 20news-19997.tar.gz data to finished the classification based on traditional machine learning methods. The [url ](http://blog.csdn.net/abcjennifer/article/details/23615947/)is here

Some of her results:
*************************
Naive Bayes
*************************
predict info:
precision:0.771
recall:0.770
f1-score:0.769
*************************
KNN
*************************
predict info:
precision:0.652
recall:0.645
f1-score:0.645
*************************
SVM
*************************
predict info:
precision:0.819
recall:0.816
f1-score:0.816
*************************
KMeans
*************************
predict info:
precision:0.289
recall:0.313
f1-score:0.266

So that's a problem :)

**Comment by MoyanZitto** [comment]
@joshuia thank you very much.

And @fchollet maybe you want to see this issue~
